### DISCUSSION TITLE
Profile Bulging

### DISCUSSION THREAD

**Discussion by RollingTechSupport** [pyroll-project/pyroll-core#258] [Ideas]
Hi Everybody, 
So I just started to work on bulging for one-step approaches.
I found two different models that are available one by B. Schmidt () and one by Y. Lee (e.g. ). 
I implemented them for a round oval pass, since they are similar in structure, they yield slightly different results. 
Maybe one of you has some measured data and can compare the results with our `pyroll-basic` configuration.
The attached notebooks contain the implementation. 

[comparison-birger-schmidt-model-lee-model.zip](https://github.com/user-attachments/files/17293299/comparison-birger-schmidt-model-lee-model.zip)

**Comment by RollingTechSupport** [comment]
And here is the model for 'ThreeRollPass'.

[byon-lee-min-model-three-roll-pass.zip](https://github.com/user-attachments/files/17325002/byon-lee-min-model-three-roll-pass.zip)

@grplan and @sag-kalibreur maybe you guys have some comparison?
Otherwise I checked it with the publications and it looks good.

**Comment by GRPlan** [comment]
Youngseog Lee did expand his approach to 3-Roll Processes in
https://www.researchgate.net/publication/318790213_An_approximate_model_to_predict_the_surface_profile_of_material_sections_in_a_3-roll_rolling_process

I did use this example for my 3-Roll Tutorial/Debugexample (see attached notebook)
[YoungseogLee_3Roll.zip](https://github.com/user-attachments/files/17328204/YoungseogLee_3Roll.zip)

Find the digitzed profiles from the article in external\PassX.kontur
In external\Example-GX.kontur are the results from FEM-simulation for comaprison

**Reply by sag-kalibreur** [reply]
@ChRen95, @GRPlan has already tested the three-roll process for round profiles and uploaded examples.

I have carried out two comparative calculations on this topic.

1. Comparison Y. Lee model - B. Schmidt model - SAG values for two rolls and round-oval pass
2. Comparison B. Schmidt model - SAG values for three roll passes and flat rolls (HEXAGON)

In the first case, the same calculation parameters result in a different perimeter and area. The calculation with the bulging model provides very good values and looks geometrically more realistic.

In the second case, the calculation results are identical for all values. The geometric representation is good for the first two passes, but the bulging is much too large for passes three to five. From the third pass at the latest, there should no longer be any bulging when using flat rolls.
The bulging model calculation only seems to insert geometry adjustments here; the calculation results of the roll passes remain identical. Again, I get an incorrect calculation with PyRoll Basic and the wusatowski_spreading. Therefore, I would recommend using only the marini_spreading for three-roll passes for the time being.

**Both notebooks can be found here:**

[comparison-birger-schmidt-model-lee-model-sag.zip](https://github.com/user-attachments/files/17397025/comparison-birger-schmidt-model-lee-model-sag.zip)

[comparison-birger-schmidt-model-three-roll-hex-sag.zip](https://github.com/user-attachments/files/17397030/comparison-birger-schmidt-model-three-roll-hex-sag.zip)

I am looking forward to feedback from the community...

**Reply by RollingTechSupport** [reply]
Thanks for the great support. 
I'm working on providing the models as a plugin but meet a problem in the architecture of PyRolL. Therefore the release will be delayed. Feel free to work with the Class version as of right now.
